### PR TITLE
feat(thin): space usage report for thin provisioned replicas

### PR DESCRIPTION
**Pull request by dsavitskiy**

### CODE REVIEW

File: io-engine/src/grpc/v1/replica.rs [hunk 1]
```diff
@@ -59,17 +59,31 @@ where
...
+}
+
 impl From<Lvol> for Replica {
     fn from(l: Lvol) -> Self {
+        let usage = l.usage();
         Self {
             name: l.name(),
             uuid: l.uuid(),
             pooluuid: l.pool_uuid(),
+            size: usage.capacity_bytes,
```
Comment (tiagolobocastro) [inline io-engine/src/grpc/v1/replica.rs:81]: l.size() and usage.capacity should be the same?

File: io-engine/src/lvs/lvs_lvol.rs [hunk 2]
```diff
@@ -81,6 +84,21 @@ impl Display for PropName {
...
+    /// Lvol size in bytes.
+    pub capacity_bytes: u64,
+    /// Amount of actually allocated disk space for this replica in bytes.
+    pub allocated_bytes: u64,
+    /// Cluster size in bytes.
+    pub cluster_size: u64,
+    /// Total number of clusters.
+    pub num_clusters: u64,
+    /// Number of actually allocated clusters.
+    pub num_allocated_clusters: u64,
```
Comment (tiagolobocastro) [inline io-engine/src/lvs/lvs_lvol.rs:99]: should we use gtter fn's rather than make these pub?

**Review by tiagolobocastro — APPROVED** [review]
Added a comment on https://github.com/openebs/spdk/commit/9824e69078cad290d9920f348a662fa11799ab3c, PTAL

**Review by mtzaurus — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**